app/streaming-alt/page.tsx

```diff
@@ -0,0 +1,28 @@
+import products from '@/lib/data/products';
+import { Suspense } from 'react';
+import { Product } from './Product';
+import {
+  RecommendedProducts,
+  RecommendedProductsSkeleton,
+} from './RecommendedProducts';
+import { Reviews, ReviewsSkeleton } from './Reviews';
+
+export default async function Posts() {
+  return (
+    <div className="space-y-16">
+      <div>
+        <Product product={products[0]} />
+      </div>
+
+      <Suspense fallback={<RecommendedProductsSkeleton />}>
+        {/* @ts-ignore */}
```

Comment:
Out of curiosity, what are all the `@ts-ignore`'s for? Does it not like living in suspense?

Reply:
TypeScript doesn't yet know about React async components. `[...] Its return type 'Promise<Element>' is not a valid JSX element.`

React/TypeScript/DefinatelyTyped teams are working on a solution.